File: Tuning/sample_code/03_prof-ex/src/fortran/calc_mass.f90
! Copyright 2024 Research Organization for Information Science and Technology
!!$---------------------------------------------------------------------
!!$ calc_mass.f90
!!$ Author: Yukihiro Ota (yota@rist.or.jp)
!!$ Last update: 1 Nov 2016
!!$---------------------------------------------------------------------

subroutine calc_mass

  use common_par,only:num_parA,num_parB,mamu_parA,mamu_parB

  use common_var,only:mass
  
  implicit none

  !! /* local variables */
  integer :: ii

  do ii = 1 , num_parA
     mass(ii) = mamu_parA
  end do
  do ii = num_parA+1 , num_parA+num_parB
     mass(ii) = mamu_parB
  end do
    
  return
end subroutine calc_mass

File: Tuning/sample_code/03_prof-ex/src/c/calc_mass.c
/* Copyright 2024 Research Organization for Information Science and Technology */
/*--------------------------------------------------------------------*/
/* calc_mass.c
   Author: Yukihiro Ota (yota@rist.or.jp)
   Last update: 29 July 2017                                          */
/*--------------------------------------------------------------------*/
#include <stdio.h>
#include "setup.h"
#include "calc.h"

void calc_mass ( InputData *IDp, AtomData *ADp )
{
  for ( int i=0; i<IDp->num_parA ; i=i+1 ) {
    ADp->mass[i] = IDp->mamu_parA;
  }
  for ( int i=IDp->num_parA; i<IDp->num_partot ; i=i+1 ) {
    ADp->mass[i] = IDp->mamu_parB;
  }
  return;
}
